crates/wallet/src/wallet/mod.rs

```diff
@@ -517,14 +514,15 @@ impl Wallet {
             .indexer
             .keychains_added
             .get(&KeychainKind::External)
-            .ok_or(LoadError::MissingDescriptor)?
+            .expect("keychain must have been added")
```

Comment:
Is it safe to `expect` the descriptor will always be recovered, or is it better to keep the `MissingDescriptor` error? since technically we were handling the error even for the external descriptor which was always present

Reply:
This is a tricky question. My original thought is that our code will always persist the right data so missing the network or either of the descriptors should be a panic. 

But on the other hand we're always loading changeset data from outside storage. And that means we could load the wrong data or possibly something outside our code could have changed the data. 

Based on that idea that at runtime we don't completely control the data we're loading, for both the internal and external keychains I'd like to throw the `LoadError::MissingDescriptor` instead of panic.

Reply:
Ok, will fix. Do you think the `MissingDescriptor` variant should hold the keychain kind so it can pass it along to `NewOrLoadError::LoadedDescriptorDoesNotMatch` ?

Reply:
Oh ya, I agree that it would be a more useful error if you include the missing `KeychainKind`.